### DISCUSSION TITLE
Modification du clavier

### DISCUSSION THREAD

**Discussion by Lfich** [Jean-Romain/tabliato#29] [Ideas]
Bonjour,

J'ai remarqué dans le code source que les claviers étaient représentés dans des CSV. Mon clavier d'accordéon est différent de ceux présents dans le dossier. Serait-il possible d'ajouter un clavier sans regénérer un exécutable (j'utilise le .exe windows).

Super tool autrement!

**Comment by Jean-Romain** [comment]
Bonjour, 

Normalement oui ca a été conçu pour pouvoir faire ca justement. En pratique non car j'ai récemment fait des changements qui font que ces fichiers sont en fait inclus avec le logiciel et ne sont pas physiquement dans des dossiers. C'est plus facile à gérer mais on perd cette fonctionnalité et je n'imaginais pas que quelqu'un en aurait vraiment besoin un jour. Mais je peux changer ca facilement pour vous permettre d'ajouter votre clavier dans une nouvelle version du logiciel.

**Comment by Jean-Romain** [comment]
Dans la version 1.3.1 c'est de nouveau possible. Voir la [FAQ](https://jean-romain.github.io/tabliato/faq.html#keyboard).

**Comment by Elouan-C** [comment]
Bonjour,

J'ai un clavier avec deux rangs et demi (Beltuna Alex 2 Sol-Do) et j'ai suivi le tutoriel dans la FAQ afin de rajouter mon clavier, mais j'obtiens ce message d'erreur : 
"Impossible de trouver le clavier monaccordeon1 dans le fichier de configuration des claviers"
<img width="502" height="131" alt="Erreur Impossible de trouver le clavier monaccordeon1 dans le fichier de configuration des claviers" src="https://github.com/user-attachments/assets/2ac5c97a-4efe-4fd2-b5ec-26f1d3ad105a" />

Je suis actuellement sur la version 1.3.6 sur Windows 11
Et voici les fichiers que j'ai modifiés / créer en suivant les infos disponibles dans la FAQ:
[assemblages_utilisateur.csv](https://github.com/user-attachments/files/23508180/assemblages_utilisateur.csv)
[clavier_beltuna_alex_2_sol_do.csv](https://github.com/user-attachments/files/23508488/clavier_beltuna_alex_2_sol_do.csv)

Je présume qu'il y a une manip à faire avec le "code" du clavier, mais je n'arrive pas a trouver le fichier de configuration des claviers qui est référencé dans le message d'erreur.

Et merci d'avoir créer cette outil, je l'ai découvert hier, et je suis entrain de mettre au propre toute mes partitions grâce a lui!

**Reply by Jean-Romain** [reply]
Je confirme le problème. Je vais regarder ca.

En attendant vous pouvez mettre votre configuration dans `assemblage.csv` ca devrait fonctionner. Cependant si vous faites une mise à jour vous perdrez votre configuration. D'où la nécessité de mettre dans `assemblage_utilisateur.csv`

Si le logiciel vous plait, rappelez vous qu'il est gratuit. Rien à payer mais si vous voulez me remercier vous pouvez partager vos partitions je les mettrais sur le site :wink:

**Reply by Jean-Romain** [reply]
Le bug est corrigé. Je produirais une version windows dans la semaine. Cependant votre fichier `clavier_beltuna_alex` n'est pas bon. Il y 4 rangs mais pas de rang 2 (`p1`, `p1''`, `p1'''`). Déjà c'est bizarre mais en plus le 4ème rang ne sera pas reconnu comme valide ; `p1'''` ne marchera pas).

Ca existe les accordéons à 4 rangs? Peut être que vous avez juste un `'` de trop.

**Reply by Elouan-C** [reply]
Merci pour votre réponse rapide, j'irai le télécharger quand il sera dispo :) 

Et effectivement, il y a un ' de trop, je n'ai que 2 rangs et demi. 
J'avais essayé de mettre directement dans le fichier assemblage.csv mais le logiciel crashait quasiment systématiquement avant d'arriver à compiler, mais je pense que ça doit être dû aux notes sur la 4e rangé que j'ai mis sans faire exprès dans le fichier de clavier.

Quand j'aurais retranscrits mes partition je n'hésiterait pas a vous en partagez ;)

**Reply by Jean-Romain** [reply]
La version 1.3.7 est en ligne. Vérifiez que votre clavier est valide et ca devrait fonctionner.
